ocs_ci/deployment/hosted_cluster.py

```diff
@@ -0,0 +1,1127 @@
+        if not cluster_names:
+            cluster_names = get_hosted_cluster_names()
+        futures = []
+        try:
+            with ThreadPoolExecutor(len(cluster_names)) as executor:
+                for name in cluster_names:
+                    futures.append(
+                        executor.submit(
+                            self.verify_hosted_ocp_cluster_from_provider,
+                            name,
+                        )
+                    )
+            return all(future.result() for future in futures)
+        except Exception as e:
+            logger.error(
+                f"Failed to verify HyperShift hosted clusters from provider: {e}"
+            )
+            return False
+
+    def download_hosted_clusters_kubeconfig_files(self):
+        """
+        Get HyperShift hosted cluster kubeconfig for multiple clusters
+        Returns:
+            list: the list of hosted cluster kubeconfig paths
+        """
+
+        if not self.hcp_binary_exists():
+            self.download_hcp_binary_with_podman()
+
+        cluster_names_to_paths = config.ENV_DATA["clusters"]
+
+        kubeconfig_paths = []
+        for name, path in cluster_names_to_paths.items():
+            kubeconfig_paths.append(self.download_hosted_cluster_kubeconfig(name, path))
+
+        return kubeconfig_paths
+
+    def deploy_multiple_odf_clients(self):
+        """
+        Deploy multiple ODF clients on hosted OCP clusters. Method tries to deploy ODF client on all hosted OCP clusters
+        If ODF was already deployed on some of the clusters, it will be skipped for those clusters.
+
+        Returns:
+            list: the list of kubeconfig paths for all hosted OCP clusters
+        """
+        kubeconfig_paths = self.update_hcp_binary()
+
+        hosted_cluster_names = get_hosted_cluster_names()
+
+        for cluster_name in hosted_cluster_names:
+            logger.info(f"Deploying ODF client on hosted OCP cluster '{cluster_name}'")
+            hosted_odf = HostedODF(cluster_name)
+            hosted_odf.do_deploy()
+
+        return kubeconfig_paths
+
+
+class HypershiftHostedOCP(HyperShiftBase, MetalLBInstaller, CNVInstaller, Deployment):
+    def __init__(self, name):
+        HyperShiftBase.__init__(self)
+        MetalLBInstaller.__init__(self)
+        CNVInstaller.__init__(self)
+        self.name = name
+        if config.ENV_DATA["clusters"].get(self.name):
+            cluster_path = config.ENV_DATA["clusters"].get(self.name)
+            self.cluster_kubeconfig = os.path.expanduser(
+                os.path.join(cluster_path, "kubeconfig")
+            )
+        else:
+            # avoid throwing an exception if the cluster path is not found for some reason
+            # this way we can continue with the next cluster
+            logger.error(
+                f"Cluster path for desired cluster with name '{self.name}' was not found in ENV_DATA.clusters"
+            )
+
+    def kubeconfig_exists(self):
```

Comment:
This function is generic and can be helpful to others. I suggest to add it to parent class `Deployment`.

Reply:
the function does not have value if `self.cluster_kubeconfig` is not an instance attribute, which is exactly the case if we move it to Deployment.
Move attribute to method argument does not make sense either, in such case let's just use `os.path.exists`

I would leave it as is for now.